kms/api-client/functions.py

```diff
@@ -0,0 +1,292 @@
+
+# Copyright 2017 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+
+import argparse
+import base64
+import io
+
+from googleapiclient import discovery
+
+
+# [START kms_create_keyring]
+def create_keyring(project_id, location, keyring):
+    """Creates a KeyRing in the given location. Potential locations include: 
+    global, asia-east1, europe-west1, us-central1, and us-east1."""
+
+    # Creates an API client for the KMS API.
+    kms_client = discovery.build('cloudkms', 'v1beta1')
+
+    # The resource name of the location associated with the KeyRing.
+    parent = 'projects/{}/locations/{}'.format(project_id, location)
+
+    # Create KeyRing
+    request = kms_client.projects().locations().keyRings().create(
+        parent=parent, body={}, keyRingId=keyring)
+    response = request.execute()
+
+    print('Created KeyRing {}.'.format(response["name"]))
+# [END kms_create_keyring]
+
+
+# [START kms_create_cryptokey]
+def create_cryptokey(project_id, location, keyring, cryptokey):
+    """Creates a CrytoKey within a KeyRing in the given location."""
+
+    # Creates an API client for the KMS API.
+    kms_client = discovery.build('cloudkms', 'v1beta1')
+
+    # The resource name of the KeyRing associated with the CryptoKey.
+    parent = 'projects/{}/locations/{}/keyRings/{}'.format(
+        project_id, location, keyring)
+
+    # Create a CryptoKey for the given KeyRing.
+    request = kms_client.projects().locations().keyRings().cryptoKeys().create(
+        parent=parent, body={"purpose": 'ENCRYPT_DECRYPT'},
+        cryptoKeyId=cryptokey)
+    response = request.execute()
+
+    print('Created CryptoKey {}.'.format(response["name"]))
+# [END kms_create_cryptokey]
+
+
+# [START kms_encrypt]
+def encrypt(project_id, location, keyring, cryptokey, plaintext_file_name,
+            encrypted_file_name):
+    """Encrypts data from a plaintext_file_name using the provided CryptoKey
+    and saves it to an encrypted_file_name so it can only be recovered with a
+    call to decrypt."""
+
+    # Creates an API client for the KMS API.
+    kms_client = discovery.build('cloudkms', 'v1beta1')
+
+    # The resource name of the CryptoKey.
+    name = 'projects/{}/locations/{}/keyRings/{}/cryptoKeys/{}'.format(
+        project_id, location, keyring, cryptokey)
+
+    # Read text from the input file.
+    with io.open(plaintext_file_name, 'rb') as plaintext_file:
+        plaintext = plaintext_file.read()
+        encoded_text = base64.b64encode(plaintext)
```

Comment:
You can de-dent everything after the call to `read()`. You no longer need the file open after getting its contents.